File: src/lib/scala/Slate.Core/src/main/scala/slate/core/tenants/Tenant.scala
/**
  * <slate_header>
  * author: Kishore Reddy
  * url: https://github.com/kishorereddy/scala-slate
  * copyright: 2016 Kishore Reddy
  * license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  * desc: a scala micro-framework
  * usage: Please refer to license on github for more info.
  * </slate_header>
  */
package slate.core.tenants

import slate.common.{DateTime, Field}
import slate.entities.core.{EntityWithId}

case class Tenant (
                      @Field("",true, 50)
                      id: Long = 0L,


                      @Field("", true, 30)
                      name:String          = "",


                      @Field("", true, 30)
                      desc:String           = "",


                      @Field("", true, 30)
                      key:String        = "",


                      @Field("", true, 30)
                      folder:String      = "",


                      @Field("", true, -1)
                      isEnabled:Boolean     = false,


                      @Field("", true, 50)
                      account:String         = "",


                      @Field("", true, 50)
                      contact:String         = "",


                      @Field("", true, 50)
                      url:String         = "",


                      @Field("",true, 50)
                      uniqueId:String = "",


                      @Field("", true, -1)
                      createdAt:DateTime  = DateTime.now(),


                      @Field("", true, -1)
                      createdBy:Int  = 0,


                      @Field("", true, -1)
                      updatedAt:DateTime  =  DateTime.now(),


                      @Field("", true, -1)
                      updatedBy:Int  = 0
                      )
  extends EntityWithId {
}

File: src/lib/kotlin/slatekit-integration/src/main/kotlin/slatekit/integration/tenants/Tenant.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package slatekit.integration.tenants

import slatekit.common.DateTime
import slatekit.common.Field
import slatekit.entities.core.EntityWithId


data class Tenant(
        @property:Field(length = 50)
        override val id: Long = 0L,


        @property:Field(length = 30)
        val name: String = "",


        @property:Field(length = 30)
        val desc: String = "",


        @property:Field(length = 30)
        val key: String = "",


        @property:Field(length = 30)
        val folder: String = "",


        @property:Field()
        val isEnabled: Boolean = false,


        @property:Field(length = 50)
        val account: String = "",


        @property:Field(length = 50)
        val contact: String = "",


        @property:Field(length = 50)
        val url: String = "",


        @property:Field(length = 50)
        val uniqueId: String = "",


        @property:Field()
        val createdAt: DateTime = DateTime.now(),


        @property:Field()
        val createdBy: Int = 0,


        @property:Field()
        val updatedAt: DateTime = DateTime.now(),


        @property:Field()
        val updatedBy: Int = 0
) : EntityWithId {
}
